src/hypothesis/extra/pandas/impl.py

```diff
@@ -0,0 +1,382 @@
+
+        min_size: the minimum number of entries in the resulting Series.
+
+        max_size: the maximum number of entries in the resulting Series.
+            If an explicit index is provided then max_size may be at most the
+            length of the index. If an index strategy is provided then whenever
+            the drawn index is too short max_size will merely be reduced.
+
+    """
+
+    index = build_index(
+        draw, index, min_size, max_size
+    )
+
+    if elements is None and dtype is None:
+        raise InvalidArgument(
+            'series require either an elements strategy or a dtype.'
+        )
+
+    size = len(index)
+
+    if dtype is not None:
+        if is_category_dtype(dtype):
+            numpy_dtype = np.dtype(object)
+            pandas_dtype = dtype
+        else:
+            numpy_dtype = np.dtype(dtype)
+            pandas_dtype = None
+
+        if elements is None:
+            elements = npst.from_dtype(numpy_dtype)
+
+        result_data = draw(npst.arrays(
+            elements=elements,
+            dtype=numpy_dtype,
+            shape=size
+        ))
+    else:
+        result_data = draw(
+            st.lists(elements, min_size=size, max_size=size)
+        )
+        if not result_data:
+            pandas_dtype = draw(
+                dtype_for_elements_strategy(elements),
+            )
+        else:
+            pandas_dtype = None
+
+    return pandas.Series(result_data, index, dtype=pandas_dtype)
+
+
+class column(object):
+    """Simple data object for describing a column in a DataFrame."""
+
+    def __init__(self, name=None, dtype=None, elements=None):
+        """Arguments:
+
+        -- name: the column name
+        -- dtype: the dtype of the column, or None to default to float. May
+            also be a strategy which will be used to generate the dtype.
+        -- elements: the strategy for generating values in this column, or None
+            to infer it from the dtype.
+
+        """
+        self.name = name
+        self.dtype = dtype
+        self.elements = elements
+        if dtype is None and elements is None:
+            raise InvalidArgument(
+                'At least one of dtype and elements must not be provided'
+            )
+        if is_category_dtype(dtype) and elements is None:
+            raise InvalidArgument(
+                'Must provide an elements strategy for category dtypes'
+            )
+
+    def __repr__(self):
+        return 'column(%r, dtype=%r, elements=%r)' % (
+            self.name, self.dtype, self.elements,
+        )
+
+
+def columns(names_or_number, dtype=None, elements=None):
+    st.check_type(
+        (Iterable,) + integer_types, names_or_number, 'names_or_number'
+    )
+    try:
+        names = list(names_or_number)
+    except TypeError:
+        names = [None] * names_or_number
+    return [
+        column(name=n, dtype=dtype, elements=elements) for n in names
+    ]
+
+
+@st.composite
+def data_frames(
```

Comment:
Adding the underscore just feels *wrong* here.  Hard to pin down why, but I'd definitely remove it.